compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -45,10 +45,16 @@ trait QuotesAndSplices {
         ctx.warning("Canceled splice directly inside a quote. '[ ${ XYZ } ] is equivalent to XYZ.", tree.sourcePos)
       case _ =>
     }
+    val qctx = inferImplicitArg(defn.QuoteContextClass.typeRef, tree.span)
```

Comment:
Inferring implicit args for implicit function types complicate existing typing rules. It's better to avoid the special case and only use it as a last resort.

Reply:
Will be required in #8281